### PR TITLE
chore(workflows): update workflow documentation

### PR DESCRIPTION
**Pull request by IliaShurygin**
# Description

This commit aims to provide some missing details, as well as improve examples to go inline with the intended way to use workflows

## Type of change

Please delete options that are not relevant.

- [x] Documentation

## Checklist:

Please delete options that are not relevant.

- [x] My commit message follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: website/docs/r/workflow.html.markdown [hunk 1]
```diff
@@ -8,53 +8,29 @@ Create and manage a workflow in New Relic.
...
-    }
-  }
-
   issues_filter {
     name = "filter-name"
     type = "FILTER"
 
     predicate {
-      attribute = "accumulations.sources"
+      attribute = "tag.team"
```
Comment (ikasovitch-newrelic) [inline website/docs/r/workflow.html.markdown:26]: i think we use tags in plural and not singular.

File: website/docs/r/workflow.html.markdown [hunk 2]
```diff
@@ -63,38 +39,86 @@ resource "newrelic_workflow" "foo" {
...
+* `destinations_enabled` - (Optional) **DEPRECATED** Whether destinations are enabled. Please use `enabled` instead:
+these two are different flags, but they are functionally identical. Defaults to true.
+* `enabled` - (Optional) Whether workflow is enabled. Defaults to true.
+* `enrichments` - (Optional) Workflow's enrichments. See [Nested enrichments blocks](#nested-enrichments-blocks) below for details.
 
 ### Nested `issues_filter` blocks
 
-Each workflow type supports a set of arguments for the `issues_filter` block:
+Issue filter defines how to identify issues that should be handled by the workflow. 
+It consists of a number of predicates.
```
Comment (ikasovitch-newrelic) [inline website/docs/r/workflow.html.markdown:56]: nit: It consists of one or more predicates.

File: website/docs/r/workflow.html.markdown [hunk 3]
```diff
@@ -63,38 +39,86 @@ resource "newrelic_workflow" "foo" {
...
+An issue must match **all** predicates to be handled by the workflow.
 
-* `name` - the filter's name.
-* `type` - the filter's type.   One of: `FILTER` or `VIEW`.
-* `predicate`
-  * `attribute` - A predicate's attribute.
-  * `operator` - A predicate's operator. One of: `CONTAINS`, `DOES_NOT_CONTAIN`, `DOES_NOT_EQUAL`, `DOES_NOT_EXACTLY_MATCH`, `ENDS_WITH`, `EQUAL`, `EXACTLY_MATCHES`, `GREATER_OR_EQUAL`, `GREATER_THAN`, `IS`, `IS_NOT`, `LESS_OR_EQUAL`, `LESS_THAN` or `STARTS_WITH` (workflows).
-  * `values` - A list of values.
+For example, we have a workflow with two predicates in the issue filter: 
+- `tag.team EXACTLY_MATCHES my_team` meaning "the issue has a tag called `team` and the tag value is set to 'my_team'"
```
Comment (ikasovitch-newrelic) [inline website/docs/r/workflow.html.markdown:61]: nit: tags.team

File: website/docs/r/workflow.html.markdown [hunk 4]
```diff
@@ -63,38 +39,86 @@ resource "newrelic_workflow" "foo" {
...
+* `name` - (Required) The name of the filter. The name only serves a cosmetic purpose and can only be seen through Terraform and GraphQL API.
+* `type` - (Required) Type of the filter. Please just set this field to `FILTER`. The field is likely to be deprecated/removed in the near future. 
+* `predicate` (Required) A condition an issue event should satisfy to be processed by the workflow 
+  * `attribute` - (Required) Issue event attribute to check
+  * `operator` - (Required) An operator to use to compare the attribute with the provided `values`. 
+One of: `CONTAINS`, `DOES_NOT_CONTAIN`, `EQUAL`, `DOES_NOT_EQUAL`, `DOES_NOT_EXACTLY_MATCH`, `STARTS_WITH`, `ENDS_WITH`,
+`EXACTLY_MATCHES`, `IS`, `IS_NOT`, `LESS_OR_EQUAL`, `LESS_THAN`, `GREATER_OR_EQUAL`, `GREATER_THAN` (see the note below)
+  * `values` - (Required) The `attribute` must match **any** of the values in this list  
+
+~> **NOTE:** We are currently working on improving documentation around for issue filters' attributes and operators.
```
Comment (ikasovitch-newrelic) [inline website/docs/r/workflow.html.markdown:77]: nice!

File: website/docs/r/workflow.html.markdown [hunk 5]
```diff
@@ -63,38 +39,86 @@ resource "newrelic_workflow" "foo" {
...
+* DONT_NOTIFY_FULLY_MUTED_ISSUES - Do not send notifications for fully muted issues, do send notifications for partially muted issues
+* DONT_NOTIFY_FULLY_OR_PARTIALLY_MUTED_ISSUES - Do not send notifications for fully or partially muted issues.
+* NOTIFY_ALL_ISSUES - Always send notifications, no matter whether the issue is muted or not
+
+### Nested `destination` blocks
+
+In order to get notified via a workflow you need two things:
+- A [notification_destination](notification_destination.html) that defines reusable credentials/settings for a 
+notification provider (e.g. webhook's `basic` credentials, Slack's OAuth credentials, PagerDuty API key, etc)
+- A [notification_channel](notification_channel.html) that describes additional notification parameters to be used
```
Comment (ikasovitch-newrelic) [inline website/docs/r/workflow.html.markdown:97]: we should use this section to explain that channels must be unique per workflow and that once a workflow is deleted we delete the relevant channels as well.

**Review by ikasovitch-newrelic — COMMENTED** [review]
Added some inline comments but looks very good other than that.

File: website/docs/r/workflow.html.markdown (hunk 1 above)
Comment (IliaShurygin) [reply website/docs/r/workflow.html.markdown:26]: I also thought so, but it is actually singular. Also, I forgot to add the required `accumulations.` prefix (will do)

File: website/docs/r/workflow.html.markdown (hunk 3 above)
Comment (IliaShurygin) [reply website/docs/r/workflow.html.markdown:61]: Not unless our UI somehow changes `tags` to `tag` when showing the filter. I'll double-check, but it probably does not.

File: website/docs/r/workflow.html.markdown (hunk 2 above)
Comment (IliaShurygin) [reply website/docs/r/workflow.html.markdown]: done

**Review by mbazhlekova — APPROVED** [review]

**Review by ikasovitch-newrelic — APPROVED** [review]
Ship it!

**Pull request merged**